common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -290,6 +291,27 @@ def student_view(self, context):
                 banner_text, special_html = special_html_view
                 if special_html and not masquerading_as_specific_student:
                     return Fragment(special_html)
+
+        # Content type gating for FBE previously only gated individual blocks
+        # This was an issue because audit learners could start a timed exam and then be unable to complete the exam
+        # even if they later upgrade because the timer would have expired.
+        # For this reason we check if content type gating is enabled for the user and gate the entire sequence in that event
+        self.gated_sequence_fragment = None
+        if self.is_time_limited and self.location.block_type == 'sequential':
+            try:
+                user = User.objects.get(id=self.runtime.user_id)
+                # importing here to avoid a circular import
+                from openedx.features.content_type_gating.models import ContentTypeGatingConfig
+                from openedx.features.content_type_gating.helpers import CONTENT_GATING_PARTITION_ID
+                self.content_type_gating_enabled = ContentTypeGatingConfig.enabled_for_enrollment(user=user, course_key=self.runtime.course_id)
+                if self.content_type_gating_enabled:
+                    # Get the content type gating locked content fragment to render for this sequence
+                    partition = self.descriptor._get_user_partition(CONTENT_GATING_PARTITION_ID)
```

Comment:
are there issues here where the partition could not be defined or the user could not have a group for that partition if a course is using some unusual configuration?